File: Treasury.Application/Contracts/V1/Requests/FinancialPagedRequest.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Treasury.Application.Contracts.V1.Requests;

public class FinancialPagedRequest : IPagedRequest
{
    public string[] Name { get; set; } = Array.Empty<string>();

    public string[] Acronym { get; set; } = Array.Empty<string>();

    public string[] Classification { get; set; } = Array.Empty<string>();

    public string[] Type { get; set; } = Array.Empty<string>();

    public bool IncludeInactive { get; set; } = false;

    public string[] Description { get; set; } = Array.Empty<string>();

    public string[] FiscalClass { get; set; } = Array.Empty<string>();

    public string[] FiscalYear { get; set; } = Array.Empty<string>();

    public string[] DotNumber { get; set; } = Array.Empty<string>();

    // Minimum Requested Amount
    public int MinimumRequestedAmount { get; set; } = 0;

    [Required]
    public int Page { get; set; } = 1;

    [Required]
    public int Rpp { get; set; } = 10;

    // TODO: Maximum Requested Amount

    // Date Range
}

File: src/main/java/edu/wpi/sga/treasury/api/contract/request/FinancialPagedRequest.java
package edu.wpi.sga.treasury.api.contract.request;

import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class FinancialPagedRequest implements PagedRequest {
    private Integer page = 1;
    private Integer resultsPerPage = 10;

    private List<String> name;
    private List<String> acronym;
    private List<String> classification;
    private List<String> type;
    private Boolean includeInactive = false;

    private List<String> description;
    private List<String> fiscalClass;
    private List<String> fiscalYear;
    private Integer minimumRequestedAmount;
}
